modules/utxo-coredao/test/unit/transaction.ts

```diff
@@ -0,0 +1,183 @@
+import * as assert from 'assert';
+import { CORE_DAO_MAINNET_CHAIN_ID, CORE_DAO_SATOSHI_PLUS_IDENTIFIER, createCoreDaoOpReturnBuffer } from '../../src';
+import { testutil } from '@bitgo/utxo-lib';
+
+describe('OP_RETURN', function () {
+  const validVersion = 2;
+  const validChainId = CORE_DAO_MAINNET_CHAIN_ID;
+  // random 20 byte buffers
+  const validDelegator = Buffer.alloc(20, testutil.getKey('wasm-possum').publicKey);
+  const validValidator = Buffer.alloc(20, testutil.getKey('possum-wasm').publicKey);
+  const validFee = 1;
+  // p2sh 2-3 script
+  const validRedeemScript = Buffer.from(
+    '522103a8295453660d5e212d4aaf82e8254e27e4c6752b2afa36e648537b644a6ca2702103099e28dd8bcb345e655b5312db0a574dded8f740eeef636cf45317bf010452982102a45b464fed0167d175d89bbc31d7ba1c288b52f64270c7eddaafa38803c5a6b553ae',
+    'hex'
+  );
+  const validTimelock = 800800;
+  it('should throw if invalid parameters are passed', function () {
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: 292,
+        chainId: validChainId,
+        delegator: validDelegator,
+        validator: validValidator,
+        fee: validFee,
+        timelock: validTimelock,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: Buffer.alloc(32, 0),
+        delegator: validDelegator,
+        validator: validValidator,
+        fee: validFee,
+        timelock: validTimelock,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: validChainId,
+        delegator: Buffer.alloc(19, 0),
+        validator: validValidator,
+        fee: validFee,
+        timelock: validTimelock,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: validChainId,
+        delegator: validDelegator,
+        validator: Buffer.alloc(19, 0),
+        fee: validFee,
+        timelock: validTimelock,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: validChainId,
+        delegator: validDelegator,
+        validator: validValidator,
+        fee: 256,
+        timelock: validTimelock,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: validChainId,
+        delegator: validDelegator,
+        validator: validValidator,
+        fee: validFee,
+        timelock: -1,
+      })
+    );
+    assert.throws(() =>
+      createCoreDaoOpReturnBuffer({
+        version: validVersion,
+        chainId: validChainId,
+        delegator: validDelegator,
+        validator: validValidator,
+        fee: validFee,
+      })
+    );
```

Comment:
hmm why tho?

Reply:
It needs to have either a redeem script or a timelock

Reply:
deserves a comment, however let's make it a type error instead

Reply:
see other comment